homeassistant/components/flume/sensor.py

```diff
@@ -0,0 +1,246 @@
+"""Sensor for displaying the number of result from Flume."""
+import base64
+from datetime import datetime, timedelta
+import json
+import logging
+
+import pytz
+import requests
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME, CONF_PASSWORD, CONF_USERNAME
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "Flume Sensor"
+
+CONF_CLIENT_ID = "client_id"
+CONF_CLIENT_SECRET = "client_secret"
+
+SCAN_INTERVAL = timedelta(minutes=1)
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Required(CONF_CLIENT_ID): cv.string,
+        vol.Required(CONF_CLIENT_SECRET): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Flume sensor."""
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    client_id = config.get(CONF_CLIENT_ID)
+    client_secret = config.get(CONF_CLIENT_SECRET)
+    time_zone = str(hass.config.time_zone)
+    name = config.get(CONF_NAME)
+
+    flume_devices = FlumeAuth(username, password, client_id, client_secret)
+
+    try:
+        for device in flume_devices._devices:
+            if device["type"] == 2:
+                flume = FlumeData(
+                    username,
+                    password,
+                    client_id,
+                    client_secret,
+                    device["id"],
+                    time_zone,
+                    SCAN_INTERVAL,
+                )
+                sensor_name = name + " " + device["id"]
+                add_entities([FlumeSensor(flume, sensor_name)], True)
+    except Exception as error:
+        _LOGGER.error("Unable to setup Flume Devices: %s", error)
+        return False
+
+
+class FlumeSensor(Entity):
+    """Representation of the Flume sensor."""
+
+    def __init__(self, Flume, name):
+        """Initialize the Flume sensor."""
+        self.Flume = Flume
+        self._name = name
+        self._state = None
+        self._unit_of_measurement = "GALLONS"
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return self._unit_of_measurement
```

Comment:
As it's not going to change, you can return the value directly instead of storing it into a variable.